src/benchmark/Akka.Cluster.Benchmarks/Serialization/ClusterMessageSerializerBenchmarks.cs

```diff
@@ -0,0 +1,137 @@
+
+    [Benchmark]
+    public byte[] Serialize_Heartbeat()
+    {
+        return _clusterMessageSerializer.ToBinary(Heartbeat);
+    }
+
+    [Benchmark]
+    public object Deserialize_Heartbeat()
+    {
+        return _clusterMessageSerializer.FromBinary(_clusterMessageSerializer.ToBinary(Heartbeat),
+            _clusterMessageSerializer.Manifest(Heartbeat));
+    }
+
+    [Benchmark]
+    public byte[] Serialize_HeartbeatRsp()
+    {
+        return _clusterMessageSerializer.ToBinary(HeartbeatRsp);
+    }
+
+    [Benchmark]
+    public object Deserialize_HeartbeatRsp()
+    {
+        return _clusterMessageSerializer.FromBinary(_clusterMessageSerializer.ToBinary(HeartbeatRsp),
+            _clusterMessageSerializer.Manifest(HeartbeatRsp));
+    }
+
+    private static readonly GossipEnvelope GossipEnvelope = new(A1.UniqueAddress, C1.UniqueAddress,
+        new Gossip(ImmutableSortedSet.Create(A1, B1, C1, D1)).Increment(new VectorClock.Node("node1"))
+            .Increment(new VectorClock.Node("node2"))
+            .Seen(A1.UniqueAddress)
+            .Seen(B1.UniqueAddress));
+    private static readonly Gossip Gossip2 = GossipEnvelope.Gossip
+        .Increment(new VectorClock.Node("node3"))
+        .Increment(new VectorClock.Node("node4"))
+        .Seen(A1.UniqueAddress).Seen(C1.UniqueAddress);
+    private static readonly Reachability Reachability = Reachability.Empty.Unreachable(A1.UniqueAddress, E1.UniqueAddress).Unreachable(B1.UniqueAddress, E1.UniqueAddress);
+
+    private static readonly Gossip Gossip3 = Gossip2.Copy(ImmutableSortedSet.Create(A1, B1, C1, D1, E1),
+        overview: Gossip2.Overview.Copy(reachability: Reachability));
+    
+    private static readonly GossipStatus GossipStatus = new(A1.UniqueAddress, Gossip3.Version);
+    private static readonly InternalClusterAction.Welcome Welcome = new(A1.UniqueAddress, Gossip3);
+    
+    [Benchmark]
+    public byte[] Serialize_GossipEnvelope()
+    {
+        return _clusterMessageSerializer.ToBinary(GossipEnvelope);
+    }
+    
+    [Benchmark]
+    public object Deserialize_GossipEnvelope()
+    {
+        return _clusterMessageSerializer.FromBinary(_clusterMessageSerializer.ToBinary(GossipEnvelope),
+            _clusterMessageSerializer.Manifest(GossipEnvelope));
+    }
+    
+    [Benchmark]
+    public byte[] Serialize_GossipStatus()
+    {
+        return _clusterMessageSerializer.ToBinary(GossipStatus);
+    }
+    
+    [Benchmark]
+    public object Deserialize_GossipStatus()
+    {
+        return _clusterMessageSerializer.FromBinary(_clusterMessageSerializer.ToBinary(GossipStatus),
+            _clusterMessageSerializer.Manifest(GossipStatus));
+    }
+    
+    [Benchmark]
+    public byte[] Serialize_Welcome()
+    {
+        return _clusterMessageSerializer.ToBinary(Welcome);
+    }
+    
+    [Benchmark]
+    public object Deserialize_Welcome()
+    {
+        return _clusterMessageSerializer.FromBinary(_clusterMessageSerializer.ToBinary(Welcome),
```

Comment:
Have to do full round-trips for all of the deserialization benchmarks, which sucks but isn't an issue if you aren't agonizing over absolute values - relative values are fine for measuring performance improvements.